src/drt/src/dr/FlexGridGraph_maze.cpp

```diff
@@ -452,10 +454,12 @@ frCost FlexGridGraph::getNextPathCost(
       }
     }
     if (isForbiddenTLen) {
+      if (drWorker_->getDRIter() >= debugMazeIter)
+        cout << "isForbiddenTLen\n";
       if (drWorker_ && drWorker_->getDRIter() >= 3) {
-        nextPathCost += ggDRCCost_ * edgeLength;
+        nextPathCost += 2*ggDRCCost_ * edgeLength;
       } else {
-        nextPathCost += ggMarkerCost_ * edgeLength;
+        nextPathCost += 2*ggMarkerCost_ * edgeLength;
```

Comment:
why double the cost?

Reply:
Because this follows the same pattern for the via-to-via viol above.

Reply:
...and I think same-net of viols are badder than diff net viols. The only hope to solve such viols is with RP, which exactly what the code is trying to do, while diff net viols may be solved to rerouting other net. I mean, a path with same-net viols is 100% sure to be wrong and have to change, but a path with diff net viols doesnt.